classes/pdf/HTMLTemplateOrderSlip.php

```diff
@@ -68,11 +68,40 @@ public function __construct(OrderSlip $order_slip, Smarty $smarty)
         // header informations
         $this->date = Tools::displayDate($this->order_slip->date_add);
         $prefix = Configuration::get('PS_CREDIT_SLIP_PREFIX', Context::getContext()->language->id);
-        $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, (int) $this->order_slip->id);
+
+        $this->title = sprintf(
+            HTMLTemplateOrderSlip::l('%1$s%2$06d'),
+            $prefix,
+            Configuration::get('PS_CREDIT_SLIP_RESET') ? $this->getOrderSlipIdForYear((int) $order_slip->id) : (int) $order_slip->id
+        );
 
         $this->shop = new Shop((int) $this->order->id_shop);
     }
 
+    /**
+     * Returns the numbering of the subscription invoice with respect to the year.
+     *
+     * @param int $id_order_slip order_slip
+     *
+     * @return int number for year
+     */
+    public function getOrderSlipIdForYear(int $id_order_slip): int
+    {
+        $date = Db::getInstance()->getValue('SELECT date_add FROM `' . _DB_PREFIX_ . 'order_slip` WHERE id_order_slip=' . $id_order_slip);
```

Comment:
So, if you select the wrong order slip, it returns 1. 
In this case, you should take the current year instead.

Reply:
@PierreRambaud 
Hello,
I have changed so that if an id does not exist it returns 0, and to be able to filter by year. If the year is not added as a parameter, it will return the number corresponding to its own year.
¿Would this solution work?